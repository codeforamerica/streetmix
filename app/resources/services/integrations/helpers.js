const { User } = require('../../../db/models')

/**
 finds the database record for the given user
 */
// might be better to do something besides return null
// we can actually probably get rid of this function if we want, but it might help
// keep some of the other code clean
const findUser = async function (userId) {
  const user = await User.findOne({ where: { auth0Id: userId } })
  if (user === null) {
    return null
  }
  return user.dataValues
}

// if the provider we're trying to find exists
// return its index value or -1 if not found
const addOrUpdateByProviderName = function (array, item) {
  const indexValue = array.findIndex(
    (_item) => _item.provider === item.provider
  )
  if (indexValue === -1) {
    array.push(item)
  } else {
    array[indexValue] = item
  }
}

/**
 given a user, check their identity data,
 depending on what data you find, add subscriber role

 TODO: make this less permissive(maybe add a status field or have logic depending on the provider), tested,
 and remove the role if we're confident we should (probably shouldn't worry about this for MVP)
 */
const syncAccountStatus = async function (userId) {
  const databaseUser = await User.findOne({ where: { auth0Id: userId } })
  if (databaseUser.identities.some((identity) => 'user_id' in identity)) {
    databaseUser.addRole('SUBSCRIBER_1')
  }
}

module.exports = { findUser, addOrUpdateByProviderName, syncAccountStatus }
