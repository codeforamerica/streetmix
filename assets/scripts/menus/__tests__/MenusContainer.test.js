/* eslint-env jest */
import React from 'react'
import { renderWithReduxAndIntl } from '../../../../test/helpers/render'
import MenusContainer from '../MenusContainer'

describe('MenusContainer', () => {
  it('renders', () => {
    const { asFragment } = renderWithReduxAndIntl(<MenusContainer />)
    expect(asFragment()).toMatchSnapshot()
  })
})
