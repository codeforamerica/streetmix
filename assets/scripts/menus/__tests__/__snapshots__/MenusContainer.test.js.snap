// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenusContainer renders 1`] = `
<DocumentFragment>
  <nav
    class="menu-bar"
  >
    <ul
      class="menu-bar-left"
    >
      <li
        class="menu-bar-title"
      >
        <img
          alt="Streemix"
          class="menu-bar-logo"
          src="test-file-stub"
        />
        <h1>
          Streetmix
        </h1>
      </li>
      <li>
        <button
          class="menu-attached"
        >
          Help
        </button>
      </li>
      <li>
        <button
          class="menu-attached"
        >
          Contact
        </button>
      </li>
      <li>
        <a
          href="https://opencollective.com/streetmix/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Donate
        </a>
      </li>
      <li>
        <a
          href="https://cottonbureau.com/people/streetmix"
          rel="noopener noreferrer"
          target="_blank"
        >
          Store
        </a>
      </li>
    </ul>
    <ul
      class="menu-bar-right"
    >
      <li>
        <a
          href="/new"
          target="_blank"
        >
          New street
        </a>
      </li>
      <li>
        <button
          class="menu-attached"
        >
          Settings
        </button>
      </li>
      <li>
        <button
          class="menu-attached"
        >
          Share
        </button>
      </li>
      <li>
        <button
          class="menu-sign-in button-secondary"
        >
          Sign in
        </button>
      </li>
    </ul>
  </nav>
  <div
    class="menus-container"
  >
    <div
      class="menu"
    >
      <a
        href="#"
      >
        About Streetmix…
      </a>
      <a
        href="#"
      >
        What’s new in Streetmix? [en]‎
      </a>
      <div
        class="help-menu-shortcuts"
      >
        <p>
          Keyboard shortcuts:
        </p>
        <table>
          <tbody>
            <tr>
              <td>
                <kbd
                  class="key"
                >
                  Backspace
                </kbd>
              </td>
              <td>
                Remove a segment you’re pointing at
                <br />
                (hold 
                <kbd
                  class="key"
                >
                  Shift
                </kbd>
                 to remove all)‎
              </td>
            </tr>
            <tr>
              <td
                dir="ltr"
              >
                <kbd
                  class="key key-icon"
                  title="Minus"
                >
                  <svg
                    class="svg-inline--fa fa-minus"
                  />
                </kbd>
                <kbd
                  class="key key-icon"
                  title="Plus"
                >
                  <svg
                    class="svg-inline--fa fa-plus"
                  />
                </kbd>
              </td>
              <td>
                Change width of a segment you’re pointing at
                <br />
                (hold 
                <kbd
                  class="key"
                >
                  Shift
                </kbd>
                 for more precision)‎
              </td>
            </tr>
            <tr>
              <td
                dir="ltr"
              >
                <kbd
                  class="key key-icon"
                  title="Left arrow"
                >
                  <svg
                    class="svg-inline--fa fa-arrow-left"
                  />
                </kbd>
                <kbd
                  class="key key-icon"
                  title="Right arrow"
                >
                  <svg
                    class="svg-inline--fa fa-arrow-right"
                  />
                </kbd>
              </td>
              <td>
                Move around the street
                <br />
                (hold 
                <kbd
                  class="key"
                >
                  Shift
                </kbd>
                 to jump to edges)‎
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <div
      class="menu"
    >
      <a
        href="https://strt.mx/discord"
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          class="svg-inline--fa menu-item-icon fa-discord"
        />
        Join Discord chat
      </a>
      <a
        href="https://github.com/streetmix/streetmix/"
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          class="svg-inline--fa menu-item-icon fa-github"
        />
        View source code on GitHub
      </a>
      <a
        href="#"
      >
        Subscribe to our newsletter
      </a>
    </div>
    <div
      class="menu"
    >
      <a
        href="https://opencollective.com/streetmix/"
        rel="noopener noreferrer"
        target="_blank"
      >
        Donate
      </a>
    </div>
    <div
      class="menu"
    >
      <h2
        class="menu-header"
        id="settings-menu-units-select"
      >
        Units
      </h2>
      <ul
        aria-labelledby="settings-menu-units-select"
        class="menu-item-group"
        role="listbox"
      >
        <li
          aria-selected="false"
          class="menu-item "
          role="option"
        >
          Metric units (meters)‎
        </li>
        <li
          aria-selected="false"
          class="menu-item "
          role="option"
        >
          Imperial units (feet)‎
        </li>
      </ul>
      <h2
        class="menu-header"
        id="settings-menu-language-select"
      >
        Language
      </h2>
      <ul
        aria-labelledby="settings-menu-language-select"
        class="menu-item-group"
        role="listbox"
      >
        <li
          aria-selected="false"
          class="menu-item"
          role="option"
        >
          <span>
            Deutsch‎
          </span>
          <span
            class="menu-item-subtext"
          >
            German
          </span>
        </li>
        <li
          aria-selected="true"
          class="menu-item menu-item-selected"
          role="option"
        >
          <svg
            class="svg-inline--fa menu-item-icon fa-check"
          />
          <span>
            English‎
          </span>
          <span
            class="menu-item-subtext"
          >
            English
          </span>
        </li>
        <li
          aria-selected="false"
          class="menu-item"
          role="option"
        >
          <span>
            Español (América Latina)‎
          </span>
          <span
            class="menu-item-subtext"
          >
            Spanish (Latin America)
          </span>
        </li>
        <li
          aria-selected="false"
          class="menu-item"
          role="option"
        >
          <span>
            Español (México)‎
          </span>
          <span
            class="menu-item-subtext"
          >
            Spanish (Mexico)
          </span>
        </li>
        <li
          aria-selected="false"
          class="menu-item"
          role="option"
        >
          <span>
            Français‎
          </span>
          <span
            class="menu-item-subtext"
          >
            French
          </span>
        </li>
        <li
          aria-selected="false"
          class="menu-item"
          role="option"
        >
          <span>
            Italiano‎
          </span>
          <span
            class="menu-item-subtext"
          >
            Italian
          </span>
        </li>
        <li
          aria-selected="false"
          class="menu-item"
          role="option"
        >
          <span>
            Polski‎
          </span>
          <span
            class="menu-item-subtext"
          >
            Polish
          </span>
        </li>
        <li
          aria-selected="false"
          class="menu-item"
          role="option"
        >
          <span>
            Português (Brasil)‎
          </span>
          <span
            class="menu-item-subtext"
          >
            Portuguese (Brazil)
          </span>
        </li>
        <li
          aria-selected="false"
          class="menu-item"
          role="option"
        >
          <span>
            Suomi‎
          </span>
          <span
            class="menu-item-subtext"
          >
            Finnish
          </span>
        </li>
        <li
          aria-selected="false"
          class="menu-item"
          role="option"
        >
          <span>
            Русский‎
          </span>
          <span
            class="menu-item-subtext"
          >
            Russian
          </span>
        </li>
        <li
          aria-selected="false"
          class="menu-item"
          role="option"
        >
          <span>
            لغة عربية‎
          </span>
          <span
            class="menu-item-subtext"
          >
            Arabic
          </span>
        </li>
        <li
          aria-selected="false"
          class="menu-item"
          role="option"
        >
          <span>
            日本語‎
          </span>
          <span
            class="menu-item-subtext"
          >
            Japanese
          </span>
        </li>
        <li
          aria-selected="false"
          class="menu-item"
          role="option"
        >
          <span>
            简体中文‎
          </span>
          <span
            class="menu-item-subtext"
          >
            Simplified Chinese
          </span>
        </li>
        <li
          aria-selected="false"
          class="menu-item"
          role="option"
        >
          <span>
            繁體中文‎
          </span>
          <span
            class="menu-item-subtext"
          >
            Traditional Chinese
          </span>
        </li>
        <li
          aria-selected="false"
          class="menu-item"
          role="option"
        >
          <span>
            한국어‎
          </span>
          <span
            class="menu-item-subtext"
          >
            Korean
          </span>
        </li>
      </ul>
    </div>
    <div
      class="menu share-menu"
    >
      <div
        class="share-sign-in-promo"
      >
        <a
          href="#"
        >
          Sign in
        </a>
         for nicer links to your streets and your personal street gallery
      </div>
      <div
        class="share-via-link-container"
      >
        Copy and paste this link to share:
        <div
          class="share-via-link-form"
        >
          <input
            class="share-via-link"
            readonly=""
            spellcheck="false"
            type="text"
            value="http://localhost/"
          />
          <button
            title="Copy to clipboard"
          >
            <svg
              class="octicon"
            />
          </button>
        </div>
      </div>
      <a
        class="share-via-twitter"
        href="https://twitter.com/intent/tweet?text=Check%20out%20this%20street%20on%20Streetmix!&url=http%3A%2F%2Flocalhost%2F"
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          class="svg-inline--fa menu-item-icon fa-twitter"
        />
        Share using Twitter
      </a>
      <a
        class="share-via-facebook"
        href="https://www.facebook.com/dialog/feed?app_id=undefined&redirect_uri=http%3A%2F%2Flocalhost%2F&link=http%3A%2F%2Flocalhost%2F&name=Unnamed%20St%20%E2%80%93%20Streetmix&description=Check%20out%20this%20street%20on%20Streetmix!"
        rel="noopener noreferrer"
        target="_blank"
      >
        <svg
          class="svg-inline--fa menu-item-icon fa-facebook-square"
        />
        Share using Facebook
      </a>
      <a
        href="#"
      >
        Print…
      </a>
      <a
        href="#"
        id="save-as-image"
      >
        Save as image…
        <span
          class="menu-item-subtext"
        >
          For including in a report, blog, etc.
        </span>
      </a>
    </div>
    <div
      class="menu identity-menu"
    >
      <a
        href=""
      >
        My streets
      </a>
      <a
        class="menu-item"
      >
        Sign out
      </a>
    </div>
  </div>
</DocumentFragment>
`;
